File: Contest_Codeforces_Round_952_Div_4/C_Good_Prefixes_naive.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jun 11 18:23:01 2024

@author: prona
"""


# def isGood(a, n):
#     for i in range(n):
#         tmp_a = a[i]
#         a[i] = 0
#         if(sum(a)==tmp_a):
#             return 1
#         a[i] = tmp_a
#     else:
#         return 0

tcase = int(input())
while(tcase):
    n = int(input())
    
    a = list(map(int, input().split()))
 
    cnt = 0    
    for i in range(n):
        for j in range(i+1):
            tmp_a = a[j]
            a[j] = 0
            if(sum(a[0:i+1])==tmp_a):
                cnt += 1
                a[j] = tmp_a
                break
            a[j] = tmp_a
        #if(isGood(a[0:i+1], i+1)):
            #cnt += 1
    
    print(cnt)
    
    tcase -= 1

File: Codeforce/Contest_Codeforces_Round_952_Div_4/C_Good_Prefixes_naive.cpp
#include "iostream"
#include "numeric"
using namespace std;


int isGood(long long int a[], long long int n){
    long long int tmp_a;
    for(int i=0; i<n; i++){
        tmp_a = a[i];
        a[i] = 0;
        if(accumulate(a, a+n, 0)==tmp_a){
            //cout<<"y "<<arraySum(a, i)<<endl;
            a[i] = tmp_a;
            return 1;
        }
        a[i] = tmp_a;
    }

    return 0;
}

int main(){

    long long int tcase, n, a[200005], res;

    cin>>tcase;
    while(tcase){

        cin>>n;
        for(int i=0; i<n; i++){
            cin>>a[i];
        }

        long long int cnt = 0;
        for(int i=0; i<n; i++){
            if(isGood(a, i+1)){
                cnt++;
                //cout<<res<<' '<<i<<endl;
            }
        }

        printf("%d\n", cnt);


        tcase--;
    }
    return 0;
}
